virtcontainers/clh.go

```diff
@@ -1202,141 +1170,188 @@ func (o *CLICmdline) Build(cmdline *CommandLine) {
 		}
 		paramBuilder.WriteString(" ")
 	}
-	cmdline.args = append(cmdline.args, strings.TrimSpace(paramBuilder.String()))
-
+	return strings.TrimSpace(paramBuilder.String())
 }
 
-//**********************************
-// The Default Builder
-//**********************************
-type DefaultCLIBuilder struct {
-	console   *CLIConsole
-	serial    *CLISerialConsole
-	apiSocket *CLIAPISocket
-	cpus      *CLICpus
-	memory    *CLIMemory
-	kernel    *CLIKernel
-	disk      *CLIDisk
-	fs        *CLIFs
-	rng       *CLIRng
-	logFile   *CLILogFile
-	vsock     *CLIVsock
-	cmdline   *CLICmdline
-	nets      *CLINets
-}
+//****************************************
+// API calls
+//****************************************
+func (clh *cloudHypervisor) isClhRunning(timeout int) (bool, error) {
+	if timeout < 0 {
```

Comment:
You could avoid checking for a valid timeout if you were using `uint` instead of `int`